crates/client-api/src/routes/energy.rs

```diff
@@ -130,13 +124,17 @@ pub async fn set_energy_balance<S: ControlStateDelegate>(
     Ok(axum::Json(response_json))
 }
 
-pub fn router<S>() -> axum::Router<S>
+pub fn router<S>(ctx: S) -> axum::Router<S>
 where
     S: NodeDelegate + ControlStateDelegate + Clone + 'static,
 {
     use axum::routing::{get, post, put};
+    let auth_middleware = axum::middleware::from_fn_with_state(ctx, auth_middleware::<S>);
     axum::Router::new()
         .route("/:identity", get(get_energy_balance::<S>))
-        .route("/:identity", post(set_energy_balance::<S>))
-        .route("/:identity", put(add_energy::<S>))
+        .route(
+            "/:identity",
+            post(set_energy_balance::<S>).route_layer(auth_middleware.clone()),
```

Comment:
wait, doesn't `auth_middleware` create an identity if it's not explicitly provided? Didn't we add `SpacetimeAuthRequired` above? Aren't these.. trying to do opposite things? Am I misunderstanding?